crates/miden-tx/src/prover/mod.rs

```diff
@@ -85,10 +96,22 @@ impl LocalTransactionProver {
             .map_err(TransactionProverError::RemoveFeeAssetFromDelta)?;
 
         // If the account is on-chain, add the update details.
-        let builder = match account.is_onchain() {
+        let builder = match partial_account.id().is_onchain() {
             true => {
-                let account_update_details = if account.is_new() {
-                    let mut account = account.clone();
+                let account_update_details = if partial_account.is_new() {
+                    let (id, nonce, code, partial_storage, partial_vault) =
```

Comment:
What is the scenario when `partial_account.id().is_onchain()` returns `false`?
Is it when the account is new?

If so, an account being onchain will always lead to `partial_account.is_new()` to return `false`, correct?

Reply:
I agree - `has_public_state()` seems to be more descriptive.